File: ios/RoutineChart/RoutineChart/Core/Utils/AppLogger.swift
//
//  AppLogger.swift
//  RoutineChart
//
//  Centralized logging utility
//

import Foundation
import OSLog

struct AppLogger {
    private static let subsystem = "com.routinechart.app"
    
    static let database = Logger(subsystem: subsystem, category: "database")
    static let sync = Logger(subsystem: subsystem, category: "sync")
    static let ui = Logger(subsystem: subsystem, category: "ui")
    static let domain = Logger(subsystem: subsystem, category: "domain")
    
    // General purpose logger
    static func log(_ message: String, category: String = "general") {
        let logger = Logger(subsystem: subsystem, category: category)
        logger.info("\(message)")
    }
    
    static func error(_ message: String, category: String = "general") {
        let logger = Logger(subsystem: subsystem, category: category)
        logger.error("\(message)")
    }
}

File: android/app/src/main/java/com/HammersTech/RoutineChart/core/utils/AppLogger.kt
package com.HammersTech.RoutineChart.core.utils

import timber.log.Timber

/**
 * Centralized logging utility using Timber
 * Provides structured logging for different app subsystems
 */
object AppLogger {
    /**
     * Initialize Timber logging
     * Should be called once in Application.onCreate()
     */
    fun init(isDebug: Boolean) {
        if (isDebug) {
            Timber.plant(Timber.DebugTree())
        }
    }

    // Subsystem loggers
    object Database {
        fun info(message: String) = Timber.tag("DB").i(message)

        fun error(
            message: String,
            error: Throwable? = null,
        ) = Timber.tag("DB").e(error, message)

        fun debug(message: String) = Timber.tag("DB").d(message)
    }

    object Sync {
        fun info(message: String) = Timber.tag("SYNC").i(message)

        fun error(
            message: String,
            error: Throwable? = null,
        ) = Timber.tag("SYNC").e(error, message)

        fun debug(message: String) = Timber.tag("SYNC").d(message)
    }

    object Auth {
        fun info(message: String) = Timber.tag("AUTH").i(message)

        fun error(
            message: String,
            error: Throwable? = null,
        ) = Timber.tag("AUTH").e(error, message)

        fun debug(message: String) = Timber.tag("AUTH").d(message)
    }

    object UseCase {
        fun info(message: String) = Timber.tag("USECASE").i(message)

        fun error(
            message: String,
            error: Throwable? = null,
        ) = Timber.tag("USECASE").e(error, message)

        fun debug(message: String) = Timber.tag("USECASE").d(message)
    }

    object UI {
        fun info(message: String) = Timber.tag("UI").i(message)

        fun error(
            message: String,
            error: Throwable? = null,
        ) = Timber.tag("UI").e(error, message)

        fun debug(message: String) = Timber.tag("UI").d(message)
    }
}
